File: trunk/fccaruby/fcca/tile.rb
class Tile

  attr_reader :x, :y
  attr_accessor :terrain

  def initialize(game, x, y)
    @game = game
    @x, @y = x, y
    @terrain = Grassland
  end

  def units
    u = []
    @game.nations.each do |nation|
      nation.units.each do |unit|
        u << unit if unit.x == @x and unit.y == @y
      end
      return u if u.length > 0
    end
    return u
  end

end

File: _old/fcca_go/tile.go
package main

import "fmt"

type Tile struct {
	terrain *Terrain
	x, y int
	game* Game
}

func newTile(game *Game, x, y int) *Tile {
	return &Tile { 
		terrain: Grassland,
		x: x,
		y: y,
		game: game,
	}
}

func (tile *Tile) units() []UnitIF {
	units := make([]UnitIF, 0)
	return units
	game := tile.game
	for _, nation := range game.nations {
		for _, unit := range nation.units {
			x, y := unit.pos()
			if x == tile.x && y == tile.y {
				units = append(units, unit)
			}
		}
	}
	return units
}
